src/evaluation.py

```diff
@@ -352,10 +354,41 @@ def match_answer_for_question(self, current_score, question, marked_answer):
         )
         return delta
 
-    def conditionally_print_explanation(self):
+    def conditionally_print_explanation(self, file_id):
         if self.should_explain_scoring:
             console.print(self.explanation_table, justify="center")
 
+        self.explanation_to_csv(file_id)
+        self.explanation_table_data_for_csv = []
+
+    # Explanation Table to CSV
+    def explanation_to_csv(self, file_id):
+        evaluation_json = open_evaluation_with_validation(self.path)
+
+        if evaluation_json["options"].get("enable_evaluation_table_to_csv", False):
+            explanation_table_header = []
+            cols = self.explanation_table.columns
+            processed_img_name, ext = file_id.split(".")
+
+            for col in cols:
+                explanation_table_header.append(col.header)
+
+            output_dir = os.path.join(
+                os.path.dirname(os.getcwd()),
+                f"OMRChecker/outputs/Evaluation/{processed_img_name}.csv",
```

Comment:
OMRChecker works on multiple directories actually, so can't use hardcoded paths. You need to access `outputs_namespace.paths` (create evaluation path there) and pass it in this function from the parent (evaluate_concatenated_response)